Correlation

ChayaSt opened the pull request:
This PR adds the following options to `normalizedFluctuationCorrelationFunctionMultiple`:

1) return D(t) - the unnormalized correlation function `<A(t) B(t)>`
2) Ability to stop accumulating the correlation function once it goes negative.

jchodera commented:
Tests are failing, but for unrelated reasons. I'll create an issue.
Can you add tests for your new features, though?

Lnaden commented:
Tests are now passing. I do have one question though. The PR here will stop the cross-correlation the first time it observes a negative correlation time. You then capture this one negative correlation time and return it as `C_n[-1]`. Is this intended behavior or did you want to capture everything up to but not including the negative correlation time?

Other than that, I think the code looks good. Do we wan to wait until tests exist for this method, or should we handle that in a separate Issue/PR?

jchodera commented:
> Tests are now passing. I do have one question though. The PR here will stop the cross-correlation the first time it observes a negative correlation time. You then capture this one negative correlation time and return it in theC_n[-1] entry. Is this intended behavior or did you want to capture everything up to but not including the negative correlation time?

This sounds like good behavior, as long as it is documented!

>  Do we wan to wait until tests exist for this method, or should we handle that in a separate Issue/PR?

Please add tests for this method!

ChayaSt commented:
I'll add tests.
On Jul 1, 2016 8:55 PM, "John Chodera" <notifications@github.com> wrote:

> Tests are now passing. I do have one question though. The PR here will
> stop the cross-correlation the first time it observes a negative
> correlation time. You then capture this one negative correlation time and
> return it in theC_n[-1] entry. Is this intended behavior or did you want to
> capture everything up to but not including the negative correlation time?
>
> This sounds like good behavior, as long as it is documented!
>
> Do we wan to wait until tests exist for this method, or should we handle
> that in a separate Issue/PR?
>
> Please add tests for this method!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/choderalab/pymbar/pull/230#issuecomment-230074647>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGSuVUWh9PngR8W11b7Juq37zC5kYVVQks5qRbbmgaJpZM4JDamp>
> .
>

ChayaSt commented:
> Tests are now passing. I do have one question though. The PR here will stop the cross-correlation the first time it observes a negative correlation time. You then capture this one negative correlation time and return it in theC_n[-1] entry. Is this intended behavior or did you want to capture everything up to but not including the negative correlation time?

For my current purpose, it doesn't really matter if I capture the one negative correlation but I changed it so that it doesn't capture it. 

I added some tests. Let me know if there's anything else I should test for.

Lnaden commented:
Looks good to me!

jchodera commented:
Thanks!
